core/storage/classifier/gae_models.py

```diff
@@ -174,3 +174,82 @@ def create(
 
         training_job_instance.put()
         return instance_id
+
+
+class ClassifierExplorationMappingModel(base_models.BaseModel):
+    """Model for mapping exploration attributes to a ClassifierDataModel.
+
+    The id of instances of this class has the form
+    {{exp_id}}.{{exp_version}}.{{state_name}}
+    """
+
+    # The ID of the classifier corresponding to the exploration attributes.
+    classifier_id = ndb.StringProperty(required=True, indexed=True)
+
+    @classmethod
+    def _generate_id(cls, exp_id, exp_version, state_name):
+        """Generates a unique ID for the Classifier Exploration Mapping of the
+        form {{exp_id}}.{{exp_version}}.{{state_name}}
+
+        Args:
+            exp_id: str. ID of the exploration.
+            exp_version: int. The exploration version at the time
+                this training job was created.
+            state_name: str. The name of the state to which the classifier
+                belongs.
+
+        Returns:
+            ID of the new Classifier Exploration Mapping instance.
+        """
+        new_id = '%s.%s.%s' % (exp_id, str(exp_version), state_name)
```

Comment:
I thought we talked about the case where state names are unicode chars. I don't see any tests for this; they should be added. (To get example data, see what the backend contains when you submit chinese/arabic/hindi characters from the frontend as part of state names.)